### PR TITLE
Rd jn update review

### PR DESCRIPTION
**Pull request by APKirstein**
Added update functionality

### CODE REVIEW

File: spec/support/factory_girl.rb [hunk 1]
```diff
@@ -15,4 +15,14 @@
     state Faker::Address.state_abbr
     zip_code Faker::Address.zip_code
   end
+
+  factory :review do
+    user_id 1
+    restaurant_id 1
```
Comment (radavis) [inline spec/support/factory_girl.rb]: Don't reference ids. Use the factories to create associated records.

```
user
restaurant
```

File: spec/features/review/user_updates_their_review_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      fill_in "Body", with: "This is a new, edited comment!"
+
+      click_button "Submit"
+
+      expect(page).to have_content("This is a new, edited comment!")
+      expect(page).to have_content("Your changes have been saved!")
+    end
+
+    let!(:user2) { FactoryGirl.create(:user) }
+    let(:review) { FactoryGirl.create(:review, restaurant_id: restaurant.id, user_id: user2.id) }
```
Comment (radavis) [inline spec/features/review/user_updates_their_review_spec.rb]: Don't use ids. 

```
let(:review) { FactoryGirl.create(:review, restaurant: restaurant, user: user2) }
```

File: spec/features/review/user_updates_their_review_spec.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      expect(page).to_not have_content("Edit Review")
+    end
+  end
+
+  context "user is not signed in" do
+    scenario "an unauthenticated user cannot update any reviews" do
+      restaurant = FactoryGirl.create(:restaurant)
+      user = FactoryGirl.create(:user)
+      FactoryGirl.create(:review, restaurant_id: restaurant.id,
+                  user_id: user.id)
```
Comment (radavis) [inline spec/features/review/user_updates_their_review_spec.rb]: Don't use ids

File: app/controllers/reviews_controller.rb [hunk 4]
```diff
@@ -4,24 +4,44 @@ def new
...
     else
       flash[:notice] = @review.errors.full_messages.join(". ")
-      redirect_to @review.restaurant
+      render "restaurants/show"
+    end
+  end
+
+  def edit
+    @review = Review.find(params[:id])
+    unless @review.creator?(current_user)
```
Comment (cmkoller) [inline app/controllers/reviews_controller.rb]: You should use a before_action to authenticate the user for this!

File: app/models/review.rb [hunk 5]
```diff
@@ -34,4 +34,8 @@ class Review < ActiveRecord::Base
     greater_than: 0,
     less_than_or_equal_to: 5
   }
+
+  def creator?(user)
+    user_id == user.id
+  end
```
Comment (cmkoller) [inline app/models/review.rb]: There should be a model test for this!

File: app/views/restaurants/show.html.erb [hunk 6]
```diff
@@ -49,28 +52,29 @@
...
+      <%= f.select :originality, Review::ORIGINALITY, :include_blank => true %>
 
-    <%= f.submit "Submit" %>
-  <% end %>
-</div>
+      <%= f.label "body" %>
+      <%= f.text_area :body %>
 
+      <%= f.submit "Submit" %>
+    <% end %>
```
Comment (cmkoller) [inline app/views/restaurants/show.html.erb]: This whole form should go into a partial!

File: app/views/reviews/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      <%= f.label "originality" %>
+      <%= f.select :originality, Review::ORIGINALITY, :include_blank => true %>
+
+      <%= f.label "body" %>
+      <%= f.text_area :body %>
+
+      <%= f.submit "Submit" %>
+    <% end %>
+  </div>
+</div>
```
Comment (cmkoller) [inline app/views/reviews/edit.html.erb]: This whole form should also go into a partial! (The same one as shown on the Show page!)

File: spec/features/review/user_updates_their_review_spec.rb [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    let!(:user2) { FactoryGirl.create(:user) }
+    let(:review) { FactoryGirl.create(
+      :review,
+      restaurant: restaurant,
+      user: user2)
+    }
+
+    scenario "an authenticated user cannot update another user's review" do
+      user = FactoryGirl.create(:user)
+      restaurant = FactoryGirl.create(:restaurant)
```
Comment (cmkoller) [inline spec/features/review/user_updates_their_review_spec.rb]: Right now there are no reviews being created - so of course no one can see an "Edit review" button! Fix this such that it's actually testing what you want it to!

File: spec/features/review/user_updates_their_review_spec.rb [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+  context "user is not signed in" do
+    scenario "an unauthenticated user cannot update any reviews" do
+      restaurant = FactoryGirl.create(:restaurant)
+      user = FactoryGirl.create(:user)
+      FactoryGirl.create(
+        :review,
+        restaurant: restaurant,
+        user: user
+        )
```
Comment (cmkoller) [inline spec/features/review/user_updates_their_review_spec.rb]: Same here - you need to first create a review for this test to be valid

File: spec/features/review/user_updates_their_review_spec.rb [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+      visit restaurant_path(id: restaurant)
+
+      expect(page).to_not have_content("Edit Review")
+    end
+  end
```
Comment (cmkoller) [inline spec/features/review/user_updates_their_review_spec.rb]: You should also try actually visiting `edit_restaurant_review_path(restaurant, review)` and making sure a message displays saying "you can't visit this page" or something. Right now, I think your page might break if an unauthenticated user tries to visit that URL directly... write a test that shows me that it won't!

File: app/models/review.rb (hunk 5 above)
Comment (PVUL) [inline app/models/review.rb]: done

File: app/controllers/reviews_controller.rb [hunk 11]
```diff
@@ -1,27 +1,49 @@
 class ReviewsController < ApplicationController
+  before_action :authenticate_user!, except: [:index]
```
Comment (cmkoller) [inline app/controllers/reviews_controller.rb]: You don't have an index action for reviews - so why are you specifying that you shouldn't authenticate user for that action?

File: config/routes.rb [hunk 12]
```diff
@@ -2,14 +2,14 @@
...
-  resources :restaurants
-
   namespace :admin do
     resources :users, only: [:index, :destroy]
     resources :restaurants, only: [:index, :destroy]
   end
 
   resources :restaurants do
-    resources :reviews, only: [:new, :create, :show]
+    resources :reviews, only: [:new, :create, :index]
```
Comment (cmkoller) [inline config/routes.rb]: You don't have an index action defined for reviews!

**Pull request merged**